presentation/components/src/presentation-components/labels/LabelsProvider.ts

```diff
@@ -66,7 +66,18 @@ export class PresentationLabelsProvider implements IPresentationLabelsProvider {
   }
 
   private async getLabelsInternal(keys: InstanceKey[]) {
-    return (await Presentation.presentation.getDisplayLabelDefinitions({ imodel: this.imodel, keys })).map((def) => def.displayValue); // WIP
+    const labelPromises = new Array<Promise<LabelDefinition[]>>();
+    const tempKeys = [...keys];
+
+    while (tempKeys.length) {
+      const batch = tempKeys.splice(0, DEFAULT_KEYS_BATCH_SIZE);
+      labelPromises.push(Presentation.presentation.getDisplayLabelDefinitions({ imodel: this.imodel, keys: batch }));
+    }
+
+    return (await Promise.all(labelPromises)).reduce((displayValues, labels) => {
+      Array.prototype.push.apply(displayValues, labels.map((def) => def.displayValue));

```

Comment:
why not `displayValues.push(...labels.map(...))`?

Reply:
Looks better, changed it